language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -398,7 +400,14 @@ public static DiagBasedPositionDetails computePositionDetails(SyntaxTree syntaxT
             }
             List<String> memberTypes = new ArrayList<>();
             for (TypeSymbol tMember : tMembers) {
-                memberTypes.add(CodeActionUtil.getPossibleType(tMember, edits, context).orElseThrow());
+                String possibleType = CodeActionUtil.getPossibleType(tMember, edits, context).orElseThrow();
+                if (tMember.typeKind() == TypeDescKind.UNION || tMember.typeKind() == TypeDescKind.SINGLETON) {
```

Comment:
@pubudu91 In this case, we have a union type like below:
```
type t1 string| int | t2;
type t2 "int" | "string" | 10;

function fn() {
   t1? t = 10;<cursor>
}
```
At cursor we suggest type guard and we need to make sure the generated type guards use proper types/values like:
```
if t is string {
} else if t is "string" {
} else if t is 10 {
}...
```
Can you confirm if there's any issue in this approach?